tmail-backend/jmap/extensions-es-v6/src/main/java/com/linagora/tmail/james/jmap/ES6EmailAddressContactSearchEngine.java

```diff
@@ -128,20 +133,29 @@ public Publisher<EmailAddressContact> autoComplete(AccountId accountId, String p
         SearchRequest request = new SearchRequest(configuration.getUserContactReadAliasName().getValue(), configuration.getDomainContactReadAliasName().getValue())
             .source(new SearchSourceBuilder()
                 .size(limit)
-                .query(QueryBuilders.boolQuery()
-                    .must(QueryBuilders.multiMatchQuery(part, EMAIL, FIRSTNAME, SURNAME))
-                    .should(QueryBuilders.termQuery(ACCOUNT_ID, accountId.getIdentifier()))
-                    .should(QueryBuilders.termQuery(DOMAIN, Username.of(accountId.getIdentifier()).getDomainPart()
-                        .map(Domain::asString)
-                        .orElse("")))
-                    .minimumShouldMatch(1)));
+                .query(buildAutoCompleteQuery(accountId, part)));
 
         return client.search(request, RequestOptions.DEFAULT)
             .flatMapIterable(searchResponse -> ImmutableList.copyOf(searchResponse.getHits().getHits()))
             .map(Throwing.function(this::extractContentFromHit).sneakyThrow())
             .distinct(EmailAddressContact::id);
     }
 
+    private QueryBuilder buildAutoCompleteQuery(AccountId accountId, String part) {
+        String[] mustFields = Optional.of(part.contains("@"))
+                .filter(FunctionalUtils.identityPredicate())
+                .map(mailPart -> EMAIL_SEARCH_FIELDS)
```

Comment:
Should we use a multimatch query when we know there's a signle field?

Reply:
sorry, I'm not understand

Reply:
I think what @chibenwa means is below, line 151, in the must predicate, you use a `multiMatchQuery` with this mustFields. So his question is does it make sense to use a multiMatch query in the case we have only one field? Shouldn't we use something more simpler in that case?